src/esportsbot/bot.py

```diff
@@ -183,11 +194,46 @@ async def initialsetup(ctx):
     if already_in_db:
         await ctx.channel.send("This server is already set up")
     else:
-        db_gateway().insert('guild_info', params={
-            'guild_id': ctx.author.guild.id})
+        db_gateway().insert('guild_info', make_guild_init_data(ctx.guild))
         await ctx.channel.send("This server has now been initialised")
 
 
+@client.event
+async def on_message(message: discord.Message):
+    if message.guild is not None and message.role_mentions:
+        db = db_gateway()
+        guildInfo = db.get('guild_info', params={'guild_id': message.guild.id})
+        roleUpdateTasks = set()
+        if guildInfo:
+            roleUpdateTasks = set()
+            for role in message.role_mentions:
+                roleData = db.get('pingable_roles', params={'role_id': role.id})
+                if roleData and not roleData[0]["on_cooldown"]:
+                    roleUpdateTasks.add(asyncio.create_task(role.edit(mentionable=False, colour=discord.Colour.darker_grey(), reason="placing pingable role on ping cooldown")))
+                    db.update('pingable_roles', {'on_cooldown': True}, {'role_id': role.id})
+                    db.update('pingable_roles', {"last_ping": datetime.now().timestamp()}, {'role_id': role.id})
+                    db.update('pingable_roles', {"ping_count": roleData[0]["ping_count"] + 1}, {'role_id': role.id})
+                    db.update('pingable_roles', {"monthly_ping_count": roleData[0]["monthly_ping_count"] + 1}, {'role_id': role.id})
+                    roleUpdateTasks.add(asyncio.create_task(client.rolePingCooldown(role, guildInfo[0]["role_ping_cooldown_seconds"])))
+                    roleUpdateTasks.add(asyncio.create_task(client.adminLog(message, {"!pingme Role Pinged": "Role: " + role.mention + "\nUser: " + message.author.mention})))
+        await client.process_commands(message)
+        if roleUpdateTasks:
+            await asyncio.wait(roleUpdateTasks)
+            for task in roleUpdateTasks:
+                if e := task.exception():
+                    traceback.print_exception(type(e), e, e.__traceback__)
+    else:
+        await client.process_commands(message)
+
```

Comment:
This is fine, just commenting to say that we will probably want to move this code and code from my feat/MusicBot into functions to make on_message cleaner as it could get quite bulky.